src/home/fetch-github/fetch-issues-preview.ts

```diff
@@ -65,16 +64,10 @@ export async function fetchIssuePreviews(): Promise<TaskNoFull[]> {
       freshIssues = publicIssues;
     }
   } catch (error) {
-    if (error instanceof RequestError && error.status === 403) {
-      await handleRateLimit(octokit, error);
+    if (!!error && typeof error === "object" && "status" in error && error.status === 403) {
+      await handleRateLimit(octokit, error as RequestError);
     } else {
-      // renderErrorInModal(error as Error, "You have been rate limited. Please login to increase your limits."); // @DEV: user another method to render the modal not as an error
-      displayPopupMessage({
-        modalHeader: "GitHub API rate limit exceeded.",
-        modalBody: "You have been rate limited. Please login to increase your limits.",
-        isError: false,
-      });
-      gitHubLoginButton?.classList.add("highlight");
+      throw error;
```

Comment:
@0x4007 Had to change this to bubble up the error, because previous code logic would display the error rate limit whatever the error was, which is wrong on a 500 network error for example.

Reply:
Well then you should catch and display the specified error for rate limiting (I think it's a 403?) 

Otherwise the raw errors don't look as presentable.

Reply:
The full logic within the `catch` clause is:
```TypeScript
    if (!!error && typeof error === "object" && "status" in error && error.status === 403) {
      await handleRateLimit(octokit, error as RequestError);
    } else {
      throw error;
    }
```

So I think we are fine. I tried and get a nice error on `403` and a generic one for the rest.